File: p3/ruby/p.rb
#!/usr/bin/ruby

def nextprime(n)
	while TRUE
		n+=1;i=2
		((n%i == 0) ? (break):(); i+=1) while i <= n
		if i == n
			return n
		end
	end
end

def run(n)
	p = 2; total = 1; bp = 2
	while total < n
		if n%p == 0
			
			total *= p
			bp = p
		end
		if (p > n) 
			return bp
		end
		p = nextprime(p)
	end
	return bp
end


puts(run(Integer(ARGV[0])))

File: p3/python/p.py
#!/usr/bin/py

import sys
def nextprime(n):
    n = n + 1
    while True:
        l = 0
        for x in range (2, n):
            l = x
            if (n%x == 0):
                break
        if (l == n-1):
            return n
        n = n + 1


def largestfactor():
    n = int(sys.argv[1])
    number = 1
    p = 2
    while number < n:
        if (n%p == 0):
            number = number * p
            bp = p
        p = nextprime(p)
        if (p > n):
            return bp
    return bp

print (largestfactor())
